scripts/py/dcstat.py

```diff
@@ -1,5 +1,19 @@
 #!/usr/bin/env python
 
+# Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+# Copyright [2016-2020] EMBL-European Bioinformatics Institute
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+
 # Requirements:
 #   Python 3.7+
```

Comment:
Default version on farm is 3.5.1, is it not too early to prepare a script for 3.7? Or has it been approved to move to 3.7 as a default already? Is it sensible to stop the script if python version is too low?

Reply:
Python 3.7.6 is installed and working on the farm already. I think new projects should start from that version unless there are good reasons not to (like libraries that absolutely need to be 3.5 compatible) otherwise we will never move forward and 3.5 is reaching is end of life anyway.

Moreover this is a standalone command line script and doesn't have dependencies (i.e no need for a virtualenv to run it) so it can be run on the farm by simply:
```
$ pyenv activate 3.7.6
$ ./dcstat.py ...
```

Reply:
Well Was not expecting this script to need a virtual env, hence my question.

Reply:
Since when running pyenv, in the background you are activating a venv, right?

Reply:
Yes, you're activating the default one for that version. What I meant is that there is no need for creating a specific virtualenv for this script as it will work with any Python 3.7+ so it can work immediately in the farm without any setup action from admins, but switching to Python 3.7.6 is required to the user.